File: benchmarks/tsb/bench_series_reflected_arith.ts
/**
 * Benchmark: series_reflected_arith — seriesRadd / seriesRsub / seriesRmul / seriesRdiv.
 * Outputs JSON: {"function": "series_reflected_arith", "mean_ms": ..., "iterations": ..., "total_ms": ...}
 */
import { Series, seriesRadd, seriesRsub, seriesRmul, seriesRdiv } from "../../src/index.ts";

const SIZE = 100_000;
const WARMUP = 5;
const ITERATIONS = 50;

const a = new Series({ data: Array.from({ length: SIZE }, (_, i) => i * 1.5) });
const b = new Series({ data: Array.from({ length: SIZE }, (_, i) => (i % 1000) + 1) });

for (let i = 0; i < WARMUP; i++) {
  seriesRadd(a, 10);
  seriesRsub(a, 1000);
  seriesRmul(a, 3);
  seriesRdiv(b, 100);
}

const start = performance.now();
for (let i = 0; i < ITERATIONS; i++) {
  seriesRadd(a, 10);
  seriesRsub(a, 1000);
  seriesRmul(a, 3);
  seriesRdiv(b, 100);
}
const total = performance.now() - start;

console.log(
  JSON.stringify({
    function: "series_reflected_arith",
    mean_ms: total / ITERATIONS,
    iterations: ITERATIONS,
    total_ms: total,
  }),
);

File: benchmarks/pandas/bench_series_reflected_arith.py
"""Benchmark: series_reflected_arith — Series.radd / rsub / rmul / rdiv."""
import json, time
import numpy as np
import pandas as pd

SIZE = 100_000
WARMUP = 5
ITERATIONS = 50

a = pd.Series(np.arange(SIZE) * 1.5)
b = pd.Series((np.arange(SIZE) % 1000) + 1.0)

for _ in range(WARMUP):
    a.radd(10)
    a.rsub(1000)
    a.rmul(3)
    b.rdiv(100)

start = time.perf_counter()
for _ in range(ITERATIONS):
    a.radd(10)
    a.rsub(1000)
    a.rmul(3)
    b.rdiv(100)
total = (time.perf_counter() - start) * 1000

print(json.dumps({
    "function": "series_reflected_arith",
    "mean_ms": total / ITERATIONS,
    "iterations": ITERATIONS,
    "total_ms": total,
}))
